store/file/close_once.go

```diff
@@ -0,0 +1,79 @@
+package file
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	eds "github.com/celestiaorg/celestia-node/share/new_eds"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ eds.AccessorCloser = (*closeOnce)(nil)
+
+var errAccessorClosed = errors.New("accessor is closed")
+
+type closeOnce struct {
+	f      eds.AccessorCloser
+	closed atomic.Bool
```

Comment:
We've amazing https://pkg.go.dev/sync#OnceFunc for that

Reply:
closeOnce is doing something more here. The motivation behind closeOnce is to prevent call of any methods on inner Accessor if it was closed.